webui/react/src/hooks/useSettingsProvider.tsx

```diff
@@ -42,17 +42,17 @@ export const SettingsProvider: React.FC<React.PropsWithChildren> = ({ children }
   });
   const checked = useObservable(authChecked);
   const [canceler] = useState(new AbortController());
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading] = useState(() => observable(true));
   const querySettings = useRef('');
-  const [settingsState, setSettingsState] = useState(() => Map<string, Settings>());
+  const [settingsState] = useState(() => observable(Map<string, Settings>()));
```

Comment:
i think we can get rid of `() =>`

Reply:
I believe we'd want to avoid calling calling functions during initialization
https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

Reply:
i got what you meant. that makes sense. thanks for clarifying